### ISSUE TITLE
tf.contrib.legacy_seq2seq.embedding_attention_seq2seq raise 'NoneType' error adding a cell with DropoutWrapper on Windows Tensorflow Version

### ISSUE THREAD

**Issue by TaoKai** [tensorflow/models#1399]
I'm building a seq2seq mnt model using tf.contrib.legacy_seq2seq.embedding_attention_seq2seq and it raised 'NoneType' error while adding a cell with DropoutWrapper. And the error occurred while using the copy.deepcopy(cell) function. And I've tried that the compilation will be passed if I remove the DropoutWrapper. My Tensorflow version is 1.1.0-cp35-cp35m-win_amd64. Part of my code is as bellow:
```python
        def single_cell():
		c=tf.contrib.rnn.GRUCell(embedding_size)
		return c
	cell=single_cell()
	if num_layers>1:
		cell=tf.contrib.rnn.MultiRNNCell([single_cell() for _ in range(num_layers)])
		cell=tf.contrib.rnn.DropoutWrapper(cell, output_keep_prob=keep_prob)
	def seq2seq_f(encoder_inputs, decoder_inputs, cell, do_decode):
		return tf.contrib.legacy_seq2seq.embedding_attention_seq2seq(
			encoder_inputs=encoder_inputs,
			decoder_inputs=decoder_inputs,
			cell=cell,
			num_encoder_symbols=src_vocab_size,
			num_decoder_symbols=target_vocab_size,
			embedding_size=embedding_size,
			output_projection=output_projection,
			feed_previous=do_decode,
			dtype='float32'
		)
	outputs, _=seq2seq_f(encode_list, decode_list, cell, forward_only)
```

**Comment by aselle** [comment]
`tf.contrib` is not officially supported, because it is temporary, testing or transient code. This code is additionally marked legacy, which means it is deprecated. @nealwu, is there some superseding package that @TaoKai should be looking at?

**Comment by nealwu** [comment]
@TaoKai you may want to look at https://www.tensorflow.org/api_docs/python/tf/contrib/seq2seq instead, since it seems to be the new version of `legacy_seq2seq`. Haven't personally used it though.

**Comment by TaoKai** [comment]
But I think the main problem is concerning the api DropoutWrapper, cuz the legacy_seq2seq could be compiled if I remove the DropoutWrapper from the cell. And the error showed that the cell cannot be 'deepcopyed' if added the DropoutWrapper. (The error raised at the deepcopy() at copy.py) Please have a check! @nealwu

**Comment by nealwu** [comment]
Sorry, I'm not sure. @ebrevdo maybe you have some ideas?

**Comment by TaoKai** [comment]
With my further test, The compilation will succeed if the *_keep_prob attribute of the rnn.DropoutWrapper is fed with a constant (0.5 or 1.0 ...), but will raise error when fed with a tf.placeholder(), and this problem seems only occur on the Windows Version. (Did not happened on Linux Version) @aselle @nealwu @ebrevdo  Please have a check of the source code......

**Comment by ebrevdo** [comment]
that's right, the dropout wrapper stores tf tensors as private member variables, unlike most other rnncells.  and deepcopy fails on these.  since the legacy seq2seq codebase is deprecated, and changes to remove the deepcopy would require a refactor that would break a lot of existing client code, i don't expect this to be resolved.  try the new seq2seq api which supports dropout wrapper without any problems.

**Comment by TaoKai** [comment]
Ok, thanks...

**Comment by TaoKai** [comment]
@nealwu hello, sorry to bother, can you please give some tutorial examples of how to use the latest tf.contrib.seq2seq api? cuz' the official Models concerning seq2seq are still using the Legacy_seq2seq, and the new api is lacking explaination, really hard to understand for me~~~ Thanks...

**Comment by nealwu** [comment]
Do you have a particular model/object you're interested in? @ebrevdo would be the best person to ask about this.
